sketch_main.cpp

```diff
@@ -175,22 +176,26 @@ int main(int argc, char *argv[]) {
 
     if (FLAGS_sketch_method.substr(FLAGS_sketch_method.size() - 2, 2) == "MH") {
         std::function<std::vector<uint64_t>(const std::vector<uint64_t> &)> sketcher;
-        MinHash<uint64_t> min_hash;
-        WeightedMinHash<uint64_t> wmin_hash;
-        OrderedMinHash<uint64_t> omin_hash;
 
         if (FLAGS_sketch_method == "MH") {
-            min_hash = MinHash<uint64_t>(kmer_word_size, FLAGS_embed_dim, "uniform");
-            sketcher = [&](const std::vector<uint64_t> &seq) { return min_hash.compute(seq); };
+            sketcher = [&](const std::vector<uint64_t> &seq) {
+                MinHash<uint64_t> min_hash(kmer_word_size, FLAGS_embed_dim, HashAlgorithm::uniform);
```

Comment:
any reason for stuffing this into the lambda? I generally prefer lambdas to be as short and simple as possible. I don' think shortening the lifetime of the sketcher here brings much of a performance improvement. :)

Reply:
Yeah, it's ugly (but fancy :P ). It's not so much about shortening lifetimes as removing all the default constructors from the `hash_base` and derived classes, so now these _need_ to be initialized.

Maybe I find a nicer way later on. One way is unique pointers but that's also overkill.